### PR TITLE
setup unit tests environment.

### PR DESCRIPTION
**Pull request by zhyd1997**
+ added [babel](https://babeljs.io/) support;
+ used [jest](https://jestjs.io/) for major test framework;
+ used [jest-mongodb](https://github.com/shelfio/jest-mongodb) for integrating with MongoDB;
+ used [supertest](https://github.com/visionmedia/supertest) for HTTP methods test.

### CODE REVIEW

File: controllers/figures.js [hunk 1]
```diff
@@ -35,11 +35,11 @@ exports.upload = asyncHandler(async (req, res, next) => {
 
 /**
  * @desc    Retrieve image
- * @route   GET /api/v1/figure/:blockKey
+ * @route   GET /api/v1/figures/:blockKey
  * @access  Private
  */
 exports.retrieve = asyncHandler(async (req, res, next) => {
-  const figure = await Figure.find({ key: req.params.blockKey });
+  const figure = await Figure.find({ blockKey: req.params.blockKey });
```
Comment (zhyd1997) [inline controllers/figures.js:42]: bug fixed.

**Comment by xavierchow** [comment]
Hi @007tom I have two questions,

* What is the purpose of returning JWT in both cookie and `response.body`?
https://github.com/zhyd1997/Eorg-Server/blob/7327fbd842d12bb9ebf40141de178c5c44b32971/controllers/auth.js#L208-L209

* I see the test cases are tight-coupled, there is a global variable `tokenNoBearerPrefix` among various test cases which makes running a single test case is not possible, do you have any idea? (I'm not asking for a fix, but just share your thoughts here)

**Comment by zhyd1997** [comment]
Hi, @xavierchow 
- I provide 2 ways of authentication for clients: `cookie` and `Bearer token`, and the server-side could get the token from either of all:
```js
  let token;

  if (
    req.headers.authorization &&
    req.headers.authorization.startsWith('Bearer')
  ) {
    // Set token from Bearer token in header
    token = req.headers.authorization.split(' ')[1];
    // Set token from cookie
  } else if (req.cookies.token) {
     token = req.cookies.token;
  }
```
As we all known, `cookie` is not supported CORS, in that case, we could use `Bearer token` instead.
- If we want to run a single test case, I think that we can do this in the following way:
```js
let db;
beforeEach(async() => {
  const connection = await mongo.connect();
  db = await connection.db();

  const users = db.collection('users');
  const mockUser = {username: "test", email: "test@test.com", password: "test123"};
  await users.insertOne(mockUser);
});

afterEach(async (done) => {
  await db.collection("users").deleteMany({});
  mongo.disconnect(done);
});

it("login successfully", () => {
  request(app)
    .post("/login")
    .auth("test@test.com", "test123") // authentication using email and password, instead of Bearer token
    ...
});

it("logout successfully", () => {
  request(app)
    .get("/logout")
    .auth("test@test.com", "test123")
    ...
});
```

**Comment by makara** [comment]
hi @007tom, Cookie is actually supported, just it requires some setup from both backend and frontend (and sometimes not so friendly if you are building an API), just saying. See https://stackoverflow.com/questions/46288437/set-cookies-for-cross-origin-requests

**Comment by zhyd1997** [comment]
@makara LGTM 👍🏿

**Pull request merged**